app/controllers/shopify_app/callback_controller.rb

```diff
@@ -8,6 +8,11 @@ class CallbackController < ActionController::Base
     def callback
       if auth_hash
         login_shop
+
+        if session[:shop_id].blank? && ShopifyApp::SessionRepository.user_storage.present?
```

Comment:
What is this catching? Is this meant for the user authentication?

Reply:
Just pushed another change after some testing/refinements that should be clearer to understand. Yes, this line is meant to kick off the 2nd oauth flow in the case user tokens are wanted.